kubernetes-discovery-client/src/main/java/io/micronaut/kubernetes/util/KubernetesUtils.java

```diff
@@ -161,6 +166,35 @@ public static PropertySource secretAsPropertySource(Secret secret) {
         return excludesFilter;
     }
 
+    /**
+     * @param client the {@link KubernetesClient}
+     * @param podLabelKeys the list of labels inside a pod
+     * @param namespace in the configuration
+     * @return the filtered labels of the current pod
+     */
+    public static Map<String, String> getPodLabels(KubernetesClient client, List<String> podLabelKeys, String namespace) {
+        Map<String, String> result = new HashMap<>();
+        // determine if we are running inside a pod. This environment variable is always been set.
+        String host = System.getenv("KUBERNETES_SERVICE_HOST");
+        if (host == null) {
+            LOG.info("Not running on k8s");
+            return Collections.emptyMap();
+        }
+
+        Pod pod = Single.fromPublisher(client.getPod(namespace, System.getenv(HOSTNAME_ENV_VARIABLE))).blockingGet();
```

Comment:
Do not block. The calling methods are already returning a `Flowable`, so you also return a `Flowable` here and merge it in the caller methods.